.github/workflows/release_and_update.yml

```diff
@@ -0,0 +1,86 @@
+name: Release and Update Dependencies
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Get latest tag
+        id: get-latest-tag
+        run: |
+          latest_tag=$(git describe --tags --abbrev=0 || echo "v0.1.0")
+          echo "latest_tag=$latest_tag" >> $GITHUB_ENV
+
+      - name: Increment version
+        id: bump-version
+        run: |
+          IFS='.' read -ra parts <<< "${{ env.latest_tag }}"
+          new_patch=$(( ${parts[2]} + 1 ))
+          new_version="v${parts[0]}.${parts[1]}.$new_patch"
```

Comment:
This only bumps the point version. I think we can do a release manually, and trigger a workflow when we have a new release. See how we publish mmtk-core to `crates.io` when we tag a release: https://github.com/mmtk/mmtk-core/blob/4ca8812607bfb0b398278e01cbd3755959c7d010/.github/workflows/cargo-publish.yml#L6